src/core/Provider.tsx

```diff
@@ -13,44 +13,51 @@ type Props = {
   settings?: Settings;
 };
 
-type State = {
-  colorScheme: ColorSchemeName;
-  reduceMotionEnabled: boolean;
-};
-
-export default class Provider extends React.Component<Props, State> {
-  state = {
-    reduceMotionEnabled: false,
-    colorScheme: Appearance?.getColorScheme() || 'light',
-  };
+const Provider = ({ ...props }: Props) => {
+  const colorSchemeName =
+    (!props.theme && Appearance?.getColorScheme()) || 'light';
 
-  async componentDidMount() {
-    AccessibilityInfo.addEventListener(
-      'reduceMotionChanged',
-      this.updateReduceMotionSettingsInfo
-    );
-    this.updateReduceMotionSettingsInfo();
-    Appearance?.addChangeListener(this.handleAppearanceChange);
-  }
+  const [reduceMotionEnabled, setReduceMotionEnabled] = React.useState<boolean>(
+    false
+  );
+  const [colorScheme, setColorScheme] = React.useState<ColorSchemeName>(
+    colorSchemeName
+  );
 
-  componentWillUnmount() {
-    AccessibilityInfo.removeEventListener(
-      'reduceMotionChanged',
-      this.updateReduceMotionSettingsInfo
-    );
-    Appearance?.removeChangeListener(this.handleAppearanceChange);
-  }
-
-  private handleAppearanceChange = (
+  const handleAppearanceChange = (
     preferences: Appearance.AppearancePreferences
   ) => {
     const { colorScheme } = preferences;
-    this.setState({ colorScheme });
+    setColorScheme(colorScheme);
   };
 
-  private getTheme = () => {
-    const { theme: providedTheme } = this.props;
-    const { reduceMotionEnabled, colorScheme } = this.state;
+  React.useEffect(() => {
+    if (!props.theme) {
+      AccessibilityInfo.addEventListener(
+        'reduceMotionChanged',
+        (reduceMotionEnabled) => setReduceMotionEnabled(reduceMotionEnabled)
```

Comment:
This creates a new function on every render. Please pass the `setReduceMotionEnabled` without creating an arrow function.

```suggestion
        setReduceMotionEnabled
```